packages/aws-cdk-lib/aws-ec2/lib/windows-versions.ts

```diff
@@ -488,991 +376,2108 @@ export enum WindowsVersion {
+  WINDOWS_SERVER_2022_ENGLISH_FULL_SQL_2019_WEB = 'Windows_Server-2022-English-Full-SQL_2019_Web',
+  WINDOWS_SERVER_2022_ENGLISH_FULL_SQL_2022_ENTERPRISE = 'Windows_Server-2022-English-Full-SQL_2022_Enterprise',
+  WINDOWS_SERVER_2022_ENGLISH_FULL_SQL_2022_EXPRESS = 'Windows_Server-2022-English-Full-SQL_2022_Express',
+  WINDOWS_SERVER_2022_ENGLISH_FULL_SQL_2022_STANDARD = 'Windows_Server-2022-English-Full-SQL_2022_Standard',
+  WINDOWS_SERVER_2022_ENGLISH_FULL_SQL_2022_WEB = 'Windows_Server-2022-English-Full-SQL_2022_Web',
+  WINDOWS_SERVER_2022_ENGLISH_STIG_CORE = 'Windows_Server-2022-English-STIG-Core',
+  WINDOWS_SERVER_2022_ENGLISH_STIG_FULL = 'Windows_Server-2022-English-STIG-Full',
+  WINDOWS_SERVER_2022_FRENCH_FULL_BASE = 'Windows_Server-2022-French-Full-Base',
+  WINDOWS_SERVER_2022_GERMAN_FULL_BASE = 'Windows_Server-2022-German-Full-Base',
+  WINDOWS_SERVER_2022_HUNGARIAN_FULL_BASE = 'Windows_Server-2022-Hungarian-Full-Base',
+  WINDOWS_SERVER_2022_ITALIAN_FULL_BASE = 'Windows_Server-2022-Italian-Full-Base',
+  WINDOWS_SERVER_2022_JAPANESE_FULL_BASE = 'Windows_Server-2022-Japanese-Full-Base',
+  WINDOWS_SERVER_2022_JAPANESE_FULL_SQL_2017_ENTERPRISE = 'Windows_Server-2022-Japanese-Full-SQL_2017_Enterprise',
+  WINDOWS_SERVER_2022_JAPANESE_FULL_SQL_2017_STANDARD = 'Windows_Server-2022-Japanese-Full-SQL_2017_Standard',
+  WINDOWS_SERVER_2022_JAPANESE_FULL_SQL_2017_WEB = 'Windows_Server-2022-Japanese-Full-SQL_2017_Web',
+  WINDOWS_SERVER_2022_JAPANESE_FULL_SQL_2019_ENTERPRISE = 'Windows_Server-2022-Japanese-Full-SQL_2019_Enterprise',
+  WINDOWS_SERVER_2022_JAPANESE_FULL_SQL_2019_STANDARD = 'Windows_Server-2022-Japanese-Full-SQL_2019_Standard',
+  WINDOWS_SERVER_2022_JAPANESE_FULL_SQL_2019_WEB = 'Windows_Server-2022-Japanese-Full-SQL_2019_Web',
+  WINDOWS_SERVER_2022_JAPANESE_FULL_SQL_2022_ENTERPRISE = 'Windows_Server-2022-Japanese-Full-SQL_2022_Enterprise',
+  WINDOWS_SERVER_2022_JAPANESE_FULL_SQL_2022_STANDARD = 'Windows_Server-2022-Japanese-Full-SQL_2022_Standard',
+  WINDOWS_SERVER_2022_JAPANESE_FULL_SQL_2022_WEB = 'Windows_Server-2022-Japanese-Full-SQL_2022_Web',
+  WINDOWS_SERVER_2022_KOREAN_FULL_BASE = 'Windows_Server-2022-Korean-Full-Base',
+  WINDOWS_SERVER_2022_POLISH_FULL_BASE = 'Windows_Server-2022-Polish-Full-Base',
+  WINDOWS_SERVER_2022_PORTUGUESE_BRAZIL_FULL_BASE = 'Windows_Server-2022-Portuguese_Brazil-Full-Base',
+  WINDOWS_SERVER_2022_PORTUGUESE_PORTUGAL_FULL_BASE = 'Windows_Server-2022-Portuguese_Portugal-Full-Base',
+  WINDOWS_SERVER_2022_RUSSIAN_FULL_BASE = 'Windows_Server-2022-Russian-Full-Base',
+  WINDOWS_SERVER_2022_SPANISH_FULL_BASE = 'Windows_Server-2022-Spanish-Full-Base',
+  WINDOWS_SERVER_2022_SWEDISH_FULL_BASE = 'Windows_Server-2022-Swedish-Full-Base',
+  WINDOWS_SERVER_2022_TURKISH_FULL_BASE = 'Windows_Server-2022-Turkish-Full-Base',
+  /** 2022 - End */
+}
+/**
+ * Specific, datestamped Windows version name.
+ */
+export enum WindowsSpecificVersion {
```

Comment:
Will this break folks currently using a timestamped version?

Reply:
The datestamped versions never worked with `WindowsVersion`, since the constructor uses `ssm:GetParameter`, looking for the `/aws/service/ami-windows-latest/<version>` key. Only non-datestamped versions keys are listed, and their corresponding value points to the latest AMI version, as shown here:
```sh
$ aws ssm get-parameters --names /aws/service/ami-windows-latest/Windows_Server-2022-English-Full-Base | jq '.Parameters[] | .Value'
"ami-03cd80cfebcbb4481"
$  aws ec2 describe-images --image-ids ami-03cd80cfebcbb4481 | jq '.Images[] | .Name'
"Windows_Server-2022-English-Full-Base-2024.03.13"
```

Reply:
Thanks for clarifying 👍🏼